### PR TITLE
5: Enables querying for Enrollment by Id

### PR DESCRIPTION
**Pull request by jlhogan**
Restructures AppQuery to pull fields from sub-query classes, idenitified
by interface marker `IAppSubQuery`. This enables separation of query
definitions into any arbitrary set of classes inheriting from
`ObjectGraphType`, as long as they are marked with the appropriate
`IAppSubquery` interface marker.

Screenshot from 2019-09-18 19-35-07

Additionally:
- Adds `.devcontainer` configuration to enable attaching to running docker-compose project in VS code, and run VSCode and extensions in that container (requires installing Remote Development extension -- @cmgiven let me know if this works for you)
- Removes Mvc service configuration which appeared superfluous; everything still works without it

### CODE REVIEW

**Comment by cmgiven** [comment]
connects #5

File: Schema/AppQuery.cs [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
-					var id = context.GetArgument<int>("id");
-					return repository.GetUserByIdAsync(id);
+					AddField(field);
 				}
-			);
+			}
 		}
 	}
+
+	public interface IAppSubQuery
```
[the remark this answers is not in the document]
Comment (jlhogan) [reply Schema/AppQuery.cs]: done

File: Schema/UserQuery.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+using GraphQL.Types;
+
+namespace Hedwig.Schema
+{
+    public class UserQuery : ObjectGraphType<object>, IAppSubQuery
+    {
+        public UserQuery(IUserRepository repository)
+        {
+            Field<UserType>(
+                "user",
```
[the remark this answers is not in the document]
Comment (jlhogan) [reply Schema/UserQuery.cs]: done

File: Schema/AppQuery.cs [hunk 3]
```diff
@@ -1,23 +1,25 @@
...
 		{
-			Field<UserType>(
-				"user",
-				arguments: new QueryArguments(
-					new QueryArgument<IdGraphType> { Name = "id" }
-				),
-				resolve: context =>
+			foreach(var subquery in appSubQueries)
+			{
+				var query = subquery as ObjectGraphType<object>;
```
[the remark this answers is not in the document]
Comment (jlhogan) [reply Schema/AppQuery.cs]: nice!! done

File: Schema/AppQuery.cs [hunk 4]
```diff
@@ -1,23 +1,25 @@
...
 using GraphQL.Types;
+using System.Collections.Generic;
 
 namespace Hedwig.Schema
 {
-	public class AppQuery : ObjectGraphType
+	public class AppQuery : ObjectGraphType<object>
 	{
-		public AppQuery(IUserRepository repository)
+		public AppQuery(IEnumerable<IAppSubQuery> appSubQueries)
```
[the remark this answers is not in the document]
Comment (jlhogan) [reply Schema/AppQuery.cs]: ty internet!!

**Comment by jlhogan** [comment]
also resolves #33

**Pull request merged**